### PR TITLE
Improved PropTypes definition in FormTab component.

### PR DESCRIPTION
**Pull request by WiXSL**

### CODE REVIEW

File: packages/ra-ui-materialui/src/form/FormTab.js [hunk 1]
```diff
@@ -73,15 +73,20 @@ const FormTab = ({
...
 FormTab.propTypes = {
+    basePath: PropTypes.string,
     className: PropTypes.string,
     contentClassName: PropTypes.string,
     children: PropTypes.node,
     intent: PropTypes.oneOf(['header', 'content']),
     hidden: PropTypes.bool,
     icon: PropTypes.element,
     label: PropTypes.string.isRequired,
+    margin: PropTypes.Margin,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/FormTab.js]: I think this is a mistake

Comment (WiXSL) [reply packages/ra-ui-materialui/src/form/FormTab.js]: I'll change it.

Comment (WiXSL) [reply packages/ra-ui-materialui/src/form/FormTab.js]: Done!

File: packages/ra-ui-materialui/src/form/FormTab.js [hunk 2]
```diff
@@ -73,15 +73,20 @@ const FormTab = ({
...
     intent: PropTypes.oneOf(['header', 'content']),
     hidden: PropTypes.bool,
     icon: PropTypes.element,
     label: PropTypes.string.isRequired,
+    margin: PropTypes.oneOf(['none', 'dense', 'normal']),
     path: PropTypes.string,
+    record: PropTypes.object,
+    resource: PropTypes.string,
     value: PropTypes.string,
+    variant: PropTypes.oneOf(['text', 'outlined', 'contained']),
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/FormTab.js]: where did you get these values from? The [material-ui doc](https://material-ui.com/api/text-field/#props) says the possible values are `standard`, `outlined`, and `filled`

Comment (WiXSL) [reply packages/ra-ui-materialui/src/form/FormTab.js]: You are right.

File: packages/ra-ui-materialui/src/form/FormTab.js [hunk 3]
```diff
@@ -73,15 +73,20 @@ const FormTab = ({
...
     intent: PropTypes.oneOf(['header', 'content']),
     hidden: PropTypes.bool,
     icon: PropTypes.element,
     label: PropTypes.string.isRequired,
+    margin: PropTypes.oneOf(['none', 'dense', 'normal']),
     path: PropTypes.string,
+    record: PropTypes.object,
+    resource: PropTypes.string,
     value: PropTypes.string,
+    variant: PropTypes.oneOf(['standar', 'outlined', 'filled']),
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/FormTab.js]: still wrong... 'standard' with a d

**Comment by fzaninotto** [comment]
The multiple comments on this PR show me that you haven't tested the code. I suggest you add unit or integration tests.

File: packages/ra-ui-materialui/src/form/FormTab.js (hunk 3 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/form/FormTab.js]: You are right I didn't test the last change. Sorry.

**Comment by WiXSL** [comment]
Ok. Could you give me a basic idea of a test regarding the check of the propTypes.

**Comment by fzaninotto** [comment]
Add a test case in the FormTab unit case where you pass all possible props to the TabbedForm (margin, variant, etc), and check that there is no warning when it runs.

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**